### DISCUSSION TITLE
DBB CLI | Deleting a Collection and all its buildResults (all dependant objects)

### DISCUSSION THREAD

**Discussion by FALLAI-Denis** [IBM/dbb#181] [Ideas]
Hi,

As part of maintenance operations on the DBB metadatastore, we had to delete Collections.
You would think that deleting a Collection would also delete all the buildResults of this Collection... but this is not the case, and apparently you have to delete the buildResults one by one...

There are also likely other logical dependencies that should be handled by the DBB API when deleting a parent object, and across multiple dependency levels.

Perhaps a "logical delete cascade" option should be included in the deletion of a Collection to delete in cascade, and upon explicit request, all associated buildResults.

Do I have to open an upgrade request on the IBM Ideas site for this?

Thanks.

**Comment by dennis-behm** [comment]
Hi @FALLAI-Denis

The Collections and the Build Groups are today loosely related. zAppBuild introduces 2 collections ( for source and link/binary dependencies) and a build group.

You don't need to delete the build results from the Metadatatstore one by one, you can make use of the **build-group delete** cmd:
https://www.ibm.com/docs/en/dbb/2.0?topic=interface-build-group-commands#build-group-delete
Have you tried this?

I think it is worth to create an idea for grouping collections and build group/results together.

Also the former WebAppCleanUp utility managed the collections and the build group individually. 

(I have enabled the discussions in ibm/dbb)
